### PR TITLE
Adds invisibility values to EVERRRRYTHING

### PR DESCRIPTION
**Pull request by Robustin**
Yea we have this great invisibility system that I can't use because almost everything is set to 0.

"Minimum" see_invisibility is at 5, so now mobs are 4, objs are 3, and turf is 2. 

Having it on a single sliding scale is still kinda lame, goggles that only let you see humans but make all turf/objs invisible would be fun... but I scaled it in the logical order. 

**This shouldn't affect anything in current gameplay** but now if, for example, you wanted a remote camera that you can't use to valid hunt but simply to check for damage as some part of, I dunno, engineering reward item, it is now far easier than trying to add overlays to all the mobs or some thing. 

I made this PR so the maintainers could yell at me for this being a horrible idea BEFORE I dumped hours into a system built around it.

### CODE REVIEW

**Comment by optimumtact** [comment]
@MrStonedOne @MrPerson

**Comment by PKPenguin321** [comment]
what are you planning just curious

**Comment by Nabski89** [comment]
Sounds like he wants to make a camera that gives all vision of the station, but disables the viewing of living mobs. That way engineers could look around for damaged areas/mouse chewed wires and know what they'll need to fix it without having to run around the entire station.

They won't be able to use it to see that clock cult base, or the power sink, but they can tell that all the doors in medbay have been welded shut.

You could also change drone vision from animals/fuzzy/letters to just NOT SEEING the things they aren't supposed to interact with.

**Comment by kevinz000** [comment]
i dont think this is a good idea as invis is used for many things like ghost fullbright and nvgs.

**Comment by Robustin** [comment]
>Invis is used for many things
>Lists the only 2 things in the game that use invis

Ho'kay

But yea NVG's only go down to 15 (by making you blind to darkness, wew) and anything else that modifies invis will go up (to 101). Most of the invis scale is useless but 0-5 is literally a var we have sitting on everything in the game that never gets used. 

When was the last time anyone recalls being unable to see something because it was invisible? Never? Oh right, because its an unused mechanic that I'm dusting off.

**Comment by Robustin** [comment]
@Nabski you're pretty close.

Though I'm considering splitting up objects into 3 and 2, with 3 being for items/weapons/etc. and 2 being for structures/machinery/bigger stuff. 

I havent decide how many objects I'd make invisible and its always possible to go through and make certain "stealthy" options have higher invis values for this mechanic. Also most cult structures/machines have an overlay that makes them invisible to mesons but I need to double check if that would work here.

**Comment by Jazaen** [comment]
>Having it on a single sliding scale is still kinda lame, goggles that only let you see humans but make all turf/objs invisible

We call them thermals.

**Comment by Robustin** [comment]
Thermals don't blind you to normal turf/objects. I'm talking full blown "everything is black except for mobs moving around", some Daredevil type shit.

**Comment by Jazaen** [comment]
Yeah, that'd be nice. And you could possibly add another invisibility level to various supernatural shit, which would 1# Make "AI can't see runes" neater and 2# Allow for "wizards and chaplains can see supernatural shit like Mime Walls"

**Comment by Robustin** [comment]
Both those mechanics would probably have to involve making specific mobs/objs have higher invis/seeing values rather than these lowered few. I might poke around more with this idea but for now I just want the foundation down before I try making invisibility more functional.

File: code/modules/mob/living/simple_animal/slime/life.dm [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 
 /mob/living/simple_animal/slime/Life()
-	set invisibility = 0
+	set invisibility = MOB_INVIS_DEFAULT
```
Comment (MrPerson) [inline code/modules/mob/living/simple_animal/slime/life.dm]: Don't do this, just remove the line outright. It shouldn't be there at all.

**Comment by MrPerson** [comment]
Invisibility is extremely limiting. Whatever you're trying to do probably isn't doable or, if it is, it would be better to just use override images in the first place. Like yes you could do spooky supernatural stuff only the chaplain can see, but then you can't have supernatural vision and darkvision since your see_invisible can only be one value. We could do pipevision but if you wanted to do another special one like wirevision either that one would have to see pipes or pipevision would have to see the wires.

Another option would be to shuffle everything onto different planes and then make the entire plane invisible/visible.

File: code/modules/mob/living/simple_animal/slime/life.dm (hunk 1 above)
Comment (Robustin) [inline code/modules/mob/living/simple_animal/slime/life.dm]: I suppose, but I mean the line had a reason to exist in the first place though right? And if the system gets more use there might be a reason to go back and say "ok maybe this mob type should be changed"

**Comment by Robustin** [comment]
@MrPerson 

Outside of overlay spamming hundreds of squares at a time, its the only way I can get stuff like my ship navigation to work without it being massively limited or otherwise used as wallhacks to hunt down antags and wall them into a 1x1 square with no floor.

**Comment by RemieRichards** [comment]
As was said, this isn't the way to go about this.

**Comment by coiax** [comment]
What /is/ the way to go about this?

**Comment by RemieRichards** [comment]
Dynamic on/off of an entire type of object (not particular instances) is a perfect task for Planes and Plane Masters, and for more maintainable.

**Comment by RemieRichards** [comment]
#25450

**Comment by Xhuis** [comment]
but how can planes fly in space when there's no air

**Pull request closed without merging**